### ISSUE TITLE
IMolecularFormula remove

### ISSUE THREAD

**Issue by lukasmur13** [cdk/cdk#1022]
Hi,

I have sum-formulas where I want to add and remove certain atoms. Using IMolecularFormula, adding can be easily done by using the add() method. Is there an equally easy method for removing molecular formulas or do I have to manipulate the string in this case?

e.g.:

IChemObjectBuilder builder = DefaultChemObjectBuilder.getInstance();
IMolecularFormula mf = MolecularFormulaManipulator.getMolecularFormula("C15H10O5", builder);
IMolecularFormula mfH = MolecularFormulaManipulator.getMolecularFormula("H", builder);
mf.add(mfH);

So if I wanted to remove one H instead, is there a method something like mf.remove(mfH)?

Thanks!

**Comment by johnmay** [comment]
removeIsotope, but you need to find the isotope first. Something like this:

```java

for (IIsotope iso : mf.isotopes()) {
  if (iso.getAtomicNumber() == IAtom.H) {
    mf.removeIsotope(iso);
    break;
  }
}
```

You might need to do it outside of the loop to avoid a concurrent modification exception.

**Comment by lukasmur13** [comment]
But this removes all hydrogen atoms, I just want to remove one. 

e.g.: 
Original: C15H10O5
-> substract 1H
New: C15H9O5

If I run the code above using isotopes, it yields:
C15O5

**Comment by lukasmur13** [comment]
I worked around it using IAtomContainer

    IAtomContainer atomContainer = MolecularFormulaManipulator.getAtomContainer(mf);
    for(IAtom atom : atomContainer.atoms()) {
        if(atom.getAtomicNumber() == IAtom.H) {
            atomContainer.removeAtom(atom);
            break;
        }
    }
    mf = MolecularFormulaManipulator.getMolecularFormula(atomContainer);

Let me know if you have a better solution, thanks!

**Comment by johnmay** [comment]
Yes it is odd that you can't seem to explicitly set the count, any ideas @egonw?

The going via an AtomContainer is overkill IMHO, perhaps a better question is why do you need to remove the hydrogen?
